src/platform/windows/mod.rs

```diff
@@ -0,0 +1,1482 @@
+#[derive(Debug)]
+struct OsIpcOutOfBandMessage {
+    channel_is_sender: Vec<bool>,
+    channel_handles: Vec<intptr_t>,
+    shmem_sizes: Vec<u64>,
+    shmem_handles: Vec<intptr_t>,
+    big_data_receiver_handle: intptr_t,
+}
+
+impl OsIpcOutOfBandMessage {
+    fn new() -> OsIpcOutOfBandMessage {
+        OsIpcOutOfBandMessage {
+            channel_is_sender: vec![],
+            channel_handles: vec![],
+            shmem_sizes: vec![],
+            shmem_handles: vec![],
+            big_data_receiver_handle: 0,
+        }
+    }
+
+    fn needs_to_be_sent(&self) -> bool {
+        !self.channel_handles.is_empty() ||
+        !self.shmem_handles.is_empty() ||
+        self.big_data_receiver_handle != 0
+    }
+
+    fn add_sender(&mut self, handle: HANDLE) {
+        self.channel_is_sender.push(true);
+        self.channel_handles.push(handle as intptr_t);
+    }
+
+    fn add_receiver(&mut self, handle: HANDLE) {
+        self.channel_is_sender.push(false);
+        self.channel_handles.push(handle as intptr_t);
+    }
+
+    fn add_shmem(&mut self, handle: HANDLE, size: usize) {
+        self.shmem_sizes.push(size as u64);
+        self.shmem_handles.push(handle as intptr_t);
+    }
+
+    fn set_big_data_receiver_handle(&mut self, handle: HANDLE) {
+        self.big_data_receiver_handle = handle as intptr_t;
+    }
+}
+
+impl serde::Serialize for OsIpcOutOfBandMessage {
+    fn serialize<S>(&self, serializer: &mut S) -> Result<(), S::Error>
+        where S: serde::Serializer
+    {
+        ((&self.channel_is_sender, &self.channel_handles,
+          &self.shmem_sizes, &self.shmem_handles,
+          &self.big_data_receiver_handle)).serialize(serializer)
+    }
+}
+
+impl serde::Deserialize for OsIpcOutOfBandMessage {
+    fn deserialize<D>(deserializer: &mut D) -> Result<OsIpcOutOfBandMessage, D::Error>
+        where D: serde::Deserializer
+    {
+        let (channel_is_sender, channel_handles,
+             shmem_sizes, shmem_handles, big_data_receiver_handle) =
+            try!(serde::Deserialize::deserialize(deserializer));
+        Ok(OsIpcOutOfBandMessage {
+            channel_is_sender: channel_is_sender, channel_handles: channel_handles,
+            shmem_sizes: shmem_sizes, shmem_handles: shmem_handles,
+            big_data_receiver_handle: big_data_receiver_handle
+        })
+    }
+}
+
+fn make_pipe_id() -> Uuid {
+    Uuid::new_v4()
+}
+
+fn make_pipe_name(pipe_id: &Uuid) -> CString {
+    CString::new(format!("\\\\.\\pipe\\rust-ipc-{}", pipe_id.to_string())).unwrap()
+}
+
+// Duplicate a given handle from this process to the target one, passing the
+// given flags to DuplicateHandle.
+//
+// Unlike win32 DuplicateHandle, this will preserve INVALID_HANDLE_VALUE (which is
+// also the pseudohandle for the current process).
+#[allow(dead_code)]
+fn dup_handle_to_process_with_flags(handle: HANDLE, other_process: HANDLE, flags: winapi::DWORD)
+                                    -> Result<HANDLE,WinError>
```

Comment:
I did the unsafe changes, as well as changing some of the handle params to `WinHandle`.  Not sure it really helped; these still need to return a raw `HANDLE`, since those go into `new()` functions that are shared with the deserialized raw handles.  So now we have signatures like:
```unsafe fn move_handle_to_process(handle: &mut WinHandle, other_process: HANDLE) -> Result<HANDLE,WinError> {```
which, I guess, is maybe better?  It does at least make it impossible to move a handle without marking a WinHandle invalid, so that's good.